### DISCUSSION TITLE
Can't Find Downloads

### DISCUSSION THREAD

**Discussion by henfredemars** [aclap-dev/video-downloadhelper#1325] [Help and Q&A]
### Your question

Hi. The tool seems to show it's downloading videos successfully, but I can't find them. Trying to open or play the video does nothing. The file path `/home/user/dwhelper` referenced in the settings doesn't exist. Where are the files going? 

Thank you for your time!

### Addon & CoApp version and details

```
version: 9.1.0.44
target: mozilla
channel: stable
lang: en-US
coapp: {"found":true,"path":"/home/user/.local/share/vdhcoapp/vdhcoapp","version":"2.0.19","new_version":false}
license: {"unneeded":true}
platform: x86-64 linux
UA: Mozilla/5.0 (X11; Linux x86_64; rv:131.0) Gecko/20100101 Firefox/131.0
{
    "used_history_button": true,
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "open_count_store": 1,
    "successfull_dl": 4,
    "download_history": {
        "__serializer_tag": "map",
        "__serializer_value": [
            {
                "__serializer_tag": "array",
                "__serializer_value": [
                    {
                        "__serializer_tag": "primitive",
                        "__serializer_value": "downloadable_5950161990396854"
                    },
                    {
                        "download_result": {
                            "inbrowser": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            },
                            "filepath": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "/home/user/dwhelper/Momiage o Shakaage o EXTRA MLP ANIMATION art - YouTube.mp4"
                            },
                            "filename": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "Momiage o Shakaage o EXTRA MLP ANIMATION art - YouTube.mp4"
                            },
                            "filedir": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "/home/user/dwhelper"
                            },
                            "qrcode": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            }
                        },
                        "page_url": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": "https://www.youtube.com/watch?v=amJdS4Lot5w&list=FL695f_Lg_qizT1K9dZ0rpvQ&index=4"
                        },
                        "timestamp": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": 1727974351930
                        }
                    }
                ]
            },
            {
                "__serializer_tag": "array",
                "__serializer_value": [
                    {
                        "__serializer_tag": "primitive",
                        "__serializer_value": "downloadable_5e600ff1-1a5f-403e-af6f-7ddf05e2d1a9"
                    },
                    {
                        "download_result": {
                            "inbrowser": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            },
                            "filepath": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "/home/user/dwhelper/Momiage o Shakaage o EXTRA MLP ANIMATION art - YouTube-01.mp4"
                            },
                            "filename": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "Momiage o Shakaage o EXTRA MLP ANIMATION art - YouTube-01.mp4"
                            },
                            "filedir": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": "/home/user/dwhelper"
                            },
                            "qrcode": {
                                "__serializer_tag": "primitive",
                                "__serializer_value": false
                            }
                        },
                        "page_url": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": "https://www.youtube.com/watch?v=amJdS4Lot5w&list=FL695f_Lg_qizT1K9dZ0rpvQ&index=4"
                        },
                        "timestamp": {
                            "__serializer_tag": "primitive",
                            "__serializer_value": 1727974507194
                        }
                    }
                ]
            }
        ]
    },
    "record_download_history": true
}
```

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

I see this in what you posted:

`"record_download_history": true`

In that case, try this:

![#01](https://github.com/user-attachments/assets/991e94b3-2196-4d37-9ebb-2c0512bf2496)

Also, I would recommend you use `Download As` instead of `Download`.

**Reply by henfredemars** [reply]
Hi! Thanks for your response. In the Download History, the Play and Show download folder options do nothing when pressed. 

Download As worked perfectly! I have no idea why Download by itself doesn't seem to place the files anywhere on the system that I can find.  

I'm not marking this as answered just yet because there could still be a bug here to be traced, but this does help me actually use the add-on.

**Reply by GrampaWildWilly** [reply]
Did you `Download As` into some place other than /home/user/dwhelper?  I know less than diddly squat about Linux but I thought that was the default directory VDH would download into.  That looks a lot like what it takes as the default on Windows, which is why I'm guessing that.  Do you have a directory named dwhelper under any directory tree on your system?

**Reply by GrampaWildWilly** [reply]
Forgot.  On my Windows system, I have VLC as my default media player.  When I click the Play button in any green entry anywhere in VDH, whether that be on the VDH menu or in the history, all it does is add that file to the VDH playlist queue.  If you don't know that's what it's doing, it can look like clicking Play did nothing.  Did that not happen for you?  As for the directory button in history, I can't explain that.  It's very weird that a download would go into the bit bucket.

**Reply by GrampaWildWilly** [reply]
I just downloaded something that completed successfully.  Using the green entry on the VDH menu, both the Play & the Show download folder buttons worked fine.  I happened to not have VLC open at the time, so the Play button caused VLC to launch but nothing played.  The downloaded file was in the VLC playlist.  Not playing but queued up is exactly what I expected.

The Show folder button on Windows is a bit anomalous, but I put that down to weirdness in Windows.  I usually have exactly one Windows file manager window open with 12 tabs.  One of those tabs is the directory into which I did this download, but at the moment I did this test, the file manager window was displaying a different tab within the window.  Clicking the Show directory button in VDH caused the file manager to gain focus, but it just showed the tab for that other directory.  I believe the file manager is simply under-featured, that there is no API for telling Windows to not just go to the file manager but to also tab over to a particular tab within the window.  I'm not entirely certain -- and I don't feel like testing this -- but if I have no file manager window open, I think VDH would actually do the right thing.

After closing VLC, I did the same tests from the VDH history.  It behaved the same.

I suspect this points to a VDH bug that could be unique to Linux.
